modules/common/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/common/LoaderMessage.scala

```diff
@@ -0,0 +1,117 @@
+
+  /** Convert to self-describing JSON */
+  def selfDescribingData: SelfDescribingData[Json] =
+    this match {
+      case _: LoaderMessage.ShreddingComplete =>
+        SelfDescribingData(LoaderMessage.ShreddingCompleteKey, (this: LoaderMessage).asJson)
+    }
+}
+
+object LoaderMessage {
+
+  val ShreddingCompleteKey: SchemaKey =
+    SchemaKey("com.snowplowanalytics.snowplow.storage.rdbloader", "shredding_complete", "jsonschema", SchemaVer.Full(1,0,0))
+
+  /** Data format for shredded data */
+  sealed trait Format extends Product with Serializable
+  object Format {
+    final case object TSV extends Format
+    final case object JSON extends Format
+    // Another options can be Parquet and InAtomic for Snowflake-like structure
+
+    implicit val loaderMessageFormatEncoder: Encoder[Format] =
+      Encoder.instance(_.toString.asJson)
+    implicit val loaderMessageFormatDecoder: Decoder[Format] =
+      Decoder.instance { c => c.as[String] match {
+        case Right("TSV") => Format.TSV.asRight
+        case Right("JSON") => Format.JSON.asRight
+        case Right(other) => DecodingFailure(s"$other is unexpected format", c.history).asLeft
+        case Left(error) => error.asLeft
+      } }
+  }
+
+  /**
+   * Set of timestamps coming from shredder
+   * @param jobStarted time shred job has started processing the batch
+   * @param jobCompleted time shred job has finished processing the batch
+   * @param min earliest collector timestamp in the batch
+   * @param max latest collector timestamp in the batch
+   */
+  final case class Timestamps(jobStarted: Instant,
+                              jobCompleted: Instant,
+                              min: Option[Instant],
+                              max: Option[Instant])
+
+  final case class ShreddedType(schemaKey: SchemaKey, format: Format)
+
+  final case class Processor(artifact: String, version: Semver)
+
+  /**
+   * Message signalling that shredder has finished and data ready to be loaded
+   * @param base root of the shredded data
+   * @param types all shredded types found in the batch
+   * @param timestamps set of auxiliary timestamps known to shredder
+   * @param processor shredder application metadata
+   */
+  final case class ShreddingComplete(base: S3.Folder,
+                                     types: List[ShreddedType],
+                                     timestamps: Timestamps,
+                                     processor: Processor) extends LoaderMessage
+
+  /** Parse raw string into self-describing JSON with [[LoaderMessage]] */
+  def fromString(s: String): Either[String, LoaderMessage] =
+    parse(s)
+      .leftMap(_.show)
+      .flatMap(json => SelfDescribingData.parse(json).leftMap(e => s"JSON message [${json.noSpaces}] is not self-describing, ${e.code}"))
+      .flatMap {
+        case SelfDescribingData(SchemaKey("com.snowplowanalytics.snowplow.storage.rdbloader", "shredding_complete", _, SchemaVer.Full(1, _, _)), data) =>
```

Comment:
Why don't we use directly `ShreddingCompleteKey` for pattern matching ? When this will be released we will support only `1-0-0` for now.

Reply:
Mostly to make sure that if we relelase `1-0-1` at some point, the Loader woudl still be ablle to parse it. In other words to make it [more robust](https://en.wikipedia.org/wiki/Robustness_principle).

Reply:
Ah yes I missed the fact that the schema can be updated without the code being updated, thanks to SemVer.